syz-cluster/controller/processor.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2024 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log"
+	"sync"
+	"time"
+
+	"github.com/google/syzkaller/syz-cluster/pkg/app"
+	"github.com/google/syzkaller/syz-cluster/pkg/blob"
+	"github.com/google/syzkaller/syz-cluster/pkg/db"
+	"github.com/google/syzkaller/syz-cluster/pkg/workflow"
+	"golang.org/x/sync/errgroup"
+)
+
+type SeriesProcessor struct {
+	blobStorage    blob.Storage
+	seriesRepo     *db.SeriesRepository
+	sessionRepo    *db.SessionRepository
+	workflows      workflow.Service
+	dbPollInterval time.Duration
+}
+
+func NewSeriesProcessor(env *app.AppEnvironment) *SeriesProcessor {
+	workflows, err := workflow.NewArgoService()
+	if err != nil {
+		app.Fatalf("failed to initialize workflows: %v", err)
+	}
+	return &SeriesProcessor{
+		blobStorage:    env.BlobStorage,
+		seriesRepo:     db.NewSeriesRepository(env.Spanner),
+		sessionRepo:    db.NewSessionRepository(env.Spanner),
+		dbPollInterval: time.Minute,
+		workflows:      workflows,
+	}
+}
+
+// Do not run more than this number of sessions in parallel.
+// TODO: it'd be different for dev and prod, make it configurable.
+const parallelWorkers = 1
```

Comment:
Make it the seriesRunner() parameter or move closer to the usage point.

Reply:
Made it configurable.